proxy/src/main/java/com/velocitypowered/proxy/connection/client/ConnectedPlayer.java

```diff
@@ -135,6 +137,8 @@ public class ConnectedPlayer implements MinecraftConnectionAssociation, Player,
 
   private static final Logger logger = LogManager.getLogger(ConnectedPlayer.class);
 
+  private static final Ratelimiter<Player> limiter = Ratelimiters.createWithMaxHitLimit(4000, 10);
```

Comment:
Personally I think that putting 4 seconds would break some uses. In my experience, for example, in servers with protections you can get to use several commands quickly when claiming terrain or configuring some things. For that kind of cases, I think it would be better to add an argument when starting velocity to be able to customize this ratelimit or a new configuration option

Reply:
It's 10 messages/commands in 4 seconds. The Minecraft server uses this same limit. I will work on adding a config option

Reply:
> 

Players with lag will always be thrown off by this feature, I think it is better that the native velocity commands (either /velocity) be cached.

Reply:
> > 
> 
> Players with lag will always be thrown off by this feature, I think it is better that the native velocity commands (either /velocity) be cached.

@xIsm4 I'm testing this feature on my personal server, `play.tecnocraft.net` with around 300 online users averagely. I haven't noticed any complaint yet.

Reply:
_This is an obvious memory leak_. For that reason alone, I'm not merging this straight away.

I'd also want to see the ability to have a customizable rate limit.